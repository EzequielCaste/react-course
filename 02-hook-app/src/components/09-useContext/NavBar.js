import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
  return (
  <nav class="navbar navbar-expand-sm navbar-dark bg-dark">
    <Link to="/" class="navbar-brand">useContext</Link>
    
    <div class="collapse navbar-collapse" id="navbarNavAltMarkup">
      <div class="navbar-nav">
        <NavLink exact activeClassName="active" to="/" className="nav-link">Home</NavLink>
        <NavLink exact activeClassName="active" to="/about" className="nav-link">About</NavLink>
        <NavLink exact activeClassName="active" to="/login" className="nav-link">Login</NavLink>
        
      </div>
    </div>
  </nav>
  )
}
