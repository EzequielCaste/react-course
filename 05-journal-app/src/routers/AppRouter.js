import React, { useEffect, useState } from "react";
import {
  BrowserRouter as Router,
  Switch, Redirect
} from "react-router-dom";
import { firebase } from '../firebase/firebase-config';


import { AuthRouter } from "./AuthRouter";
import { JournalScreen } from "../components/journal/JournalScreen";
import { useDispatch } from "react-redux";
import { login } from "../actions/auth";
import { PublicRoute } from "./PublicRoutes";
import { PrivateRoute } from "./PrivateRoutes";
import { loadNotes } from "../helpers/loadNotes";
import { setNotes } from "../actions/notes";

export const AppRouter = () => {

  const dispatch = useDispatch();

  const [checking, setChecking] = useState(true);

  const [isLoggedIn, setIsLoggedIn] = useState(false);

  useEffect(() => {   

    firebase.auth().onAuthStateChanged( async(user) => {
      
      if( user?.uid ) {
        dispatch( login(user.uid, user.displayName ) );
        setIsLoggedIn( true );

        const notes = await loadNotes( user.uid );

        dispatch( setNotes(notes) )

      } else {
        setIsLoggedIn( false );
      }

      setChecking(false);

    });

  }, [dispatch, setChecking, setIsLoggedIn])

  if ( checking ) {
    return (
      <h1>Loading...</h1>
    )
  }

  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute
            path="/auth"
            component={ AuthRouter }
            isLoggedIn={ isLoggedIn }
          />

           <PrivateRoute
            exact 
            path="/" 
            component={ JournalScreen } 
            isLoggedIn={ isLoggedIn }
          />

          <Redirect to="/auth/login" />

        </Switch>
      </div>
    </Router>
  )
}
