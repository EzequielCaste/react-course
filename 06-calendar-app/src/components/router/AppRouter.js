import React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";
import { CalendarScreen } from "../calendar/CalendarScreen";
import { LoginScreen } from "../auth/LoginScreen";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Switch>
          <Route exact path="/" component={CalendarScreen} />
          <Route exact path="/login" component={LoginScreen} />
        </Switch>
      </div>
    </Router>
  )
}
