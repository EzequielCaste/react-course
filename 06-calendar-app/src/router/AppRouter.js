import React, { useEffect } from "react";
import {
  BrowserRouter as Router,
  Switch
} from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";



import { PublicRoute } from "./PublicRoutes";
import { PrivateRoute } from "./PrivateRoutes";
import { LoginScreen } from "../components/auth/LoginScreen";
import { CalendarScreen } from "../components/calendar/CalendarScreen";
import { startChecking } from "../actions/auth";

export const AppRouter = () => {

  const dispatch = useDispatch();

  const { checking, uid } = useSelector( state => state.auth );

   useEffect(() => {

    dispatch( startChecking() );

  }, [dispatch]);


  if ( checking ) {
    return <h5>Wait...</h5>
  }

  return (
    <Router>
      <div>
        <Switch>
          <PrivateRoute 
            exact
            path="/" 
            component={CalendarScreen}  
            isLoggedIn={ !!uid  }
          />

          <PublicRoute
            exact
            path="/login"
            component={LoginScreen} 
            isLoggedIn={ !!uid }
          />
        </Switch>
      </div>
    </Router>
  )
}
