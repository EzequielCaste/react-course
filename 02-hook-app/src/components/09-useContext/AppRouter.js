import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,  
  Redirect
} from "react-router-dom";
import { NavBar } from './NavBar';
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <Switch>
          <Route path="/about" component={ AboutScreen } />
          <Route path="/login" component={ LoginScreen } />
          <Route path="/" component={ HomeScreen } />

          {/* <Route component={ HomeScreen } /> */}
          <Redirect to="/" />
        </Switch>
      </div>      
    </Router>
  )
}
