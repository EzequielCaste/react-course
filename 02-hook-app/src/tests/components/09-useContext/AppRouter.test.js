import React from 'react'
import { mount } from 'enzyme';
import '@testing-library/jest-dom'
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en <AppRouter />', () => {

  const user = {
    id: 1,
    name: 'Eze'
  }
  const wrapper = mount(
  <UserContext.Provider value={{
    user
  }}>
     <AppRouter />);
  </UserContext.Provider>);

  test('debe mostrarse correctamente', () => {

    expect( wrapper ).toMatchSnapshot();

  })

})
