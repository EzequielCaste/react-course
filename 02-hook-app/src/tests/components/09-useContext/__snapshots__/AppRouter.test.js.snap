// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe mostrarse correctamente 1`] = `
Array [
  <div
    className="container"
  >
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            activeclassname="active"
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <a
            activeclassname="active"
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            activeclassname="active"
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </nav>
    <div>
      <h1>
        Home Screen
      </h1>
      <hr />
      <pre>
        {
   "id": 1,
   "name": "Eze"
}
      </pre>
    </div>
  </div>,
  ");",
]
`;
